packages/mobile/src/dappkit/DappKitSignTxScreen.tsx

```diff
@@ -50,22 +44,22 @@ class DappKitSignTxScreen extends React.Component<Props, State> {
     this.setState({ request })
   }
 
+  getRequest = () => {
+    return this.props.route.params.dappKitRequest
+  }
+
   linkBack = () => {
-    if (!this.state.request) {
-      return
-    }
+    const request = this.getRequest()
 
-    navigateHome({ dispatchAfterNavigate: requestTxSignature(this.state.request!) })
+    navigateHome({ onAfterNavigate: () => this.props.requestTxSignature(request) })
```

Comment:
Once we navigate home, this screen will be dismounted. So won't this reference to `this.props` be invalid? Or at least prevent it from getting garbage collected?

Reply:
It will keep the reference from being garbage collected until `onAfterNavigate` is called by `requestAnimationFrame`. Which is fine since it's done in less than a second.